app/services/piece-access-level-service.js

```diff
@@ -62,4 +66,32 @@ export default class PieceAccessLevelService extends Service {
       previousPiece = await previousPiece.previousPiece;
     }
   }
+
+  /*
+   * This method is used to update the access levels of the pieces of a subcase's treatments. When a subcase has
+   * been updated to being confidential, all the pieces connected to it through agenda-item-treatments get the
+   * "Ministerraad" access level.
+   */
+  async updateDecisionsAccessLevelOfSubcase(subcase) {
+    await subcase.preEditOrSaveCheck();
+
+    if (!subcase.hasDirtyAttributes) {
+      return;
+    }
+
+    let { confidential: [, confidential] = [] } = subcase.changedAttributes();
+    if (!confidential) {
+      return;
+    }
+
+    const ministerraad = await this.store.findRecordByUri('access-level', CONSTANTS.ACCESS_LEVELS.MINISTERRAAD);
+
+    const treatments = await subcase.treatments;
```

Comment:
You could also include a filter on accessLevels so that you only get pieces that need changing

Reply:
don't think filter on access level would work, because we can't query for `not ministerraad` and neither for `intern secretarie or intern regering or intern overheid or publiek`, given that we are working with URIs, but I did use a query instead of accessing through relations.